### ISSUE TITLE
Specify remote connection_string un-exposed to Git (e.g. as environment variable)

### ISSUE THREAD

**Issue by casparjespersen** [iterative/dvc#2217]
Currently, the effect of assigning a connection string to an (Azure) remote is, that the `.dvc/config` file is updated with the actual connection string. This means that it will be exposed to Git, which is unlucky if you care about your data security.

I would suggest a fix where dvc automatically searches for the connection string in an environment variable (or a .env-file, or similar), e.g. `DVC_REMOTE_{NAME}`. This makes it more secure, and would also allow it to work with modern CI tools.

**Comment by pared** [comment]
My 5 cents about implementation:
For user convenience, maybe we should  override `dvc remote modify myremote connection_string my-connection-string` logic so that it creates some automatically ignored config file?

**Comment by casparjespersen** [comment]
@pared Right, that would also make sense. And then CI systems would simply need to do run that command prior to running whatever commands are required. I think that would be a good solution.

**Comment by efiop** [comment]
@pared great point!

@casparjespersen Or you could simply use `AZURE_STORAGE_CONNECTION_STRING` env var, that is supported already :slightly_smiling_face:

**Comment by pared** [comment]
@efiop 
Good point, didn't know about that. 
Even though, I would at least warn user about possible implications of using `dvc remote modify {} connection_string` for azure.

**Comment by casparjespersen** [comment]
@efiop `AZURE_STORAGE_CONNECTION_STRING` is, as I understand it, supported due to how the Azure Storage SDK works. This also means that it does not support two separate remotes. Most use-cases would only use a single remote, I guess, but if DVC is supposed to work with multiple remotes, then this is not a viable solution :-)

**Comment by efiop** [comment]
@casparjespersen Great point! So we have a `--local` config `.dvc/config.local` that is added to the gitignore, so you don't track it with it and so won't expose it. This is pretty much what @pared suggested, but it is already implemented :) E.g.
```
dvc remote modify myremote connection_string my-connection-string --local
```

**Comment by casparjespersen** [comment]
@efiop If that is already supported, then that is awesome. I guess the only steps to take then is to

- Warn about possible implications if not using `--local` (as @pared also suggested), and
- Update the documentation to make people aware of `--local` :-)

**Comment by efiop** [comment]
@casparjespersen Moved this issue to dvc.org, we'll adjust our docs. Thanks for the feedback! :slightly_smiling_face:
